### PR TITLE
Add CropMirrorNormalize 3D support

### PR DESCRIPTION
**Pull request by jantonguirao**
Signed-off-by: Joaquin Anton <janton@nvidia.com>

#### Why we need this PR?
Need to support 3D inputs in CropMirrorNormalize

#### What happened in this PR?
Modified CropMirrorNormalize operator to support crop arguments in the third spatial dimension (`depth`)

**JIRA TASK**: [DALI-1035]

### CODE REVIEW

**Comment by jantonguirao** [comment]
!build

File: dali/test/python/test_operator_crop_mirror_normalize.py [hunk 1]
```diff
@@ -142,23 +142,47 @@ def define_graph(self):
...
+
     assert H >= crop_h and W >= crop_w
 
     start_y = int(np.round(np.float32(crop_y) * np.float32(H - crop_h)))
     end_y = start_y + crop_h
     start_x = int(np.round(np.float32(crop_x) * np.float32(W - crop_w)))
     end_x = start_x + crop_w
+    if input_layout.count('D') > 0:
+        assert D >= crop_d
+        start_z = int(np.round(np.float32(crop_z) * np.float32(D - crop_d)))
```
Comment (JanuszL) [inline dali/test/python/test_operator_crop_mirror_normalize.py]: Please check https://github.com/NVIDIA/DALI/blob/master/dali/test/python/test_operator_crop_mirror_normalize.py#L158.

Comment (jantonguirao) [reply dali/test/python/test_operator_crop_mirror_normalize.py]: done

File: dali/pipeline/operators/fused/crop_mirror_normalize.h [hunk 2]
```diff
@@ -54,16 +44,20 @@ kernels::SliceFlipNormalizePermutePadArgs<Dims> GetKernelArgs(
...
-  for (std::size_t d = 0; d < Dims; d++) {
+  for (size_t d = 0; d < Dims; d++) {
     args.anchor[d] = slice_anchor[d];
   }
 
+  int channel_dim_idx = ImageLayoutInfo::ChannelDimIndex(input_layout);
+  assert(channel_dim_idx >= 0);
   if (pad_output) {
-    args.padded_shape[channels_dim(input_layout)] = 4;
+    args.padded_shape[channel_dim_idx] = 4;
```
Comment (mzient) [inline dali/pipeline/operators/fused/crop_mirror_normalize.h]: Pad to the smallest power of 2 that can accommodate existing number of channels.
Change the doc string (it says it pads to a multiple of 4, but that's not really necessary (?)).

Comment (jantonguirao) [reply dali/pipeline/operators/fused/crop_mirror_normalize.h]: done

**Comment by jantonguirao** [comment]
!build

**Pull request merged**